core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/util/SignerMethodResolver.java

```diff
@@ -56,7 +57,7 @@ public static SigningMethod resolveSigningMethodUsed(Signer signer, ExecutionAtt
                 signingMethod = SigningMethod.PROTOCOL_STREAMING_SIGNING_AUTH;
             } else if (isProtocolBasedUnsigned(signer, executionAttributes)) {
                 signingMethod = SigningMethod.PROTOCOL_BASED_UNSIGNED;
-            } else if (isAnonymous(credentials)) {
+            } else if (isAnonymous(credentials) || signer instanceof NoOpSigner) {
```

Comment:
Per discussion with @joviegas, NoOpSigner should be treated as anonymous credentials